dask_glm/algorithms.py

```diff
@@ -406,8 +406,8 @@ def proximal_grad(X, y, regularizer='l1', lamduh=0.1, family=Logistic,
         # Compute the step size
         lf = func
         for ii in range(100):
-            beta = regularizer.proximal_operator(obeta - stepSize * gradient, stepSize * lamduh)
-            step = obeta - beta
+            beta = regularizer.proximal_operator(- stepSize * gradient + obeta, stepSize * lamduh)
+            step = - beta + obeta
```

Comment:
I'm curious, why this change?

Reply:
When `obeta` is a `cupy array`, `cupy's` operators `+/-` can't handle a `dask array` as the second operands in those cases.

Reply:
Maybe you could do `obeta = da.from_array(obeta)` before the computation?

Reply:
It doesn't matter either way.  Either way is fine.  Can we report this upstream to cupy?  Presumably they need to return NotImplemented somwhere?

Reply:
> It doesn't matter either way. Either way is fine.

I know, just providing a more Dask-like solution.

> Can we report this upstream to cupy? Presumably they need to return NotImplemented somwhere?

I don't know exactly what error we get, but would reporting it change anything? Would returning `NotImplemented` really make things much clearer to bother?

Reply:
If a Python object doesn't know what to do with another operand in a binary operator, it can return NotIimplemented (or maybe it's raise?) and then Python will try the other order (calling `other.__rsub__(self)`)

Reply:
Thanks @mrocklin , I didn't know of that. I filed https://github.com/cupy/cupy/issues/4118 to start the discussion on the CuPy side, I'm not sure but maybe there's a technical reason not to do that, so let's wait to hear from maintainers.